matrix-meetings-bot/src/util/migrateMeetingTime.ts

```diff
@@ -26,31 +28,38 @@ export interface IMeetingTime {
 }
 
 /**
- * Changes meeting time from start and end time model to calendar model if external rrule is provided.
+ * Changes meeting time from start and end time model to calendar model.
  * @param meetingTime start, end, calendar data
  * @param externalRrule external rrule if available
+ * @param existingCalendar meeting current calendar if exists, is used if migration is needed
  */
 export function migrateMeetingTime(
   meetingTime: IMeetingTime,
   externalRrule?: string,
-): IMeetingTime {
-  if (
```

Comment:
I wouldn't overcomplicate this. If anyone is using the API variant with the old deprecated model, we can only support a single calendar entry with no special features or overrides. Overrides etc. are only possible if they send us a prepared `calendar` array.

So plain and simple:

```ts
export function migrateMeetingTime(
  meetingTime: IMeetingTime,
  externalRrule?: string | undefined,
  existingCalendar?: CalendarEntryDto[],
): CalendarEntryDto[] {
  if (meetingTime.start_time && meetingTime.end_time && !meetingTime.calendar) {
    return [
      {
        uid: existingCalendar?.[0]?.uid ?? uuiv4(),
        dtstart: formatICalDate(DateTime.fromISO(meetingTime.start_time)),
        dtend: formatICalDate(DateTime.fromISO(meetingTime.end_time)),
        rrule: externalRrule,
      },
    ];
  }

  return meetingTime.calendar ?? [];
}

```

Reply:
I forgot that OX meetings cannot be changed in meetings widget. I would prefer to throw an Error instead of returning an empty array, but is fine to me, thanks!

Reply:
yes that would be fine. This should never happen, but TypeScript was not able to see that.

Reply:
changed a bit to throw an Error